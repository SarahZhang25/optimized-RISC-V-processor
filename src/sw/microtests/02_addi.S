// This tests that your processor handles ADDI correctly.
.text
start:
    addi x1, x0, 1
    addi x2, x0, 2
    addi x2, x0, 0x7ff
    addi x4, x0, -1
    addi x5, x1, 1
    addi x6, x2, -1
    addi x7, x3, 1
    addi x8, x4, 1
    addi x9, x0, 0x100
    addi x10, x0, 0x200
    addi x11, x0, 0x400
    addi x12, x0, -2048
    unimp
