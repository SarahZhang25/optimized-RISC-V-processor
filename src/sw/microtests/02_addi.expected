Dumping the state of the processor
pc = 0x00000030
x0 = 0x00000000
x1 = 0x00000001
x2 = 0x000007ff
x3 = 0x00000000
x4 = 0xffffffff
x5 = 0x00000002
x6 = 0x000007fe
x7 = 0x00000001
x8 = 0x00000000
x9 = 0x00000100
x10 = 0x00000200
x11 = 0x00000400
x12 = 0xfffff800
x13 = 0x00000000
x14 = 0x00000000
x15 = 0x00000000
x16 = 0x00000000
x17 = 0x00000000
x18 = 0x00000000
x19 = 0x00000000
x20 = 0x00000000
x21 = 0x00000000
x22 = 0x00000000
x23 = 0x00000000
x24 = 0x00000000
x25 = 0x00000000
x26 = 0x00000000
x27 = 0x00000000
x28 = 0x00000000
x29 = 0x00000000
x30 = 0x00000000
x31 = 0x00000000
Quitting simulation.
